packages/cli/src/util/getHttpOperations.ts

```diff
@@ -0,0 +1,43 @@
+import { transformOas2Operation, transformOas3Operation } from '@stoplight/http-spec';
+import { parse } from '@stoplight/yaml';
+import axios from 'axios';
+import * as fs from 'fs';
+import { defaults, flatten, get, keys, map, uniq } from 'lodash';
+import { EOL } from 'os';
+import { httpAndFileResolver } from '../resolvers/http-and-file';
+
+export default async function getHttpOperations(spec: string) {
+  const fileContent = spec.match(/^https?:\/\//)
+    ? (await axios.get(spec)).data
+    : fs.readFileSync(spec, { encoding: 'utf8' });
+  const parsedContent = parse(fileContent);
+  const { result: resolvedContent, errors } = await httpAndFileResolver.resolve(parsedContent);
+
+  if (errors.length) {
+    const uniqueErrors = uniq(errors.map(error => error.message)).join(EOL);
+    throw new Error(uniqueErrors);
+  }
+
+  const isOas2 = get(parsedContent, 'swagger');
+
+  const transformOperationFn = isOas2 ? transformOas2Operation : transformOas3Operation;
+
+  const paths = keys(get(resolvedContent, 'paths'));
+  const methods = ['get', 'post', 'put', 'delete', 'options', 'head', 'patch', 'trace'];
+  return flatten(
+    map(paths, path =>
+      keys(get(resolvedContent, ['paths', path]))
+        .filter(k => methods.includes(k))
+        .map(method =>
+          defaults(
+            transformOperationFn({
+              document: resolvedContent,
+              path,
+              method,
+            }),
+            { servers: resolvedContent.servers },
+          ),
+        ),
+    ),
+  );
+}
```

Comment:
I see you removed loader tests but haven't covered this file yet. Are you planning to do it?

Reply:
Yes, I should definitely write some test for this function. I just wanted at least another set of eyes confirming that the implementation is correct, since I'm not 100% sure. I mean harness tests pass which is great but still…